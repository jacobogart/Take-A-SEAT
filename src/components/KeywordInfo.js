import React, { Component } from 'react';
import '../css/KeywordInfo.scss';

class KeywordInfo extends Component {
  render() {
    let { details } = this.props.keyword;
    return (
      <div>
        <p>{details}</p>
        {this.props.isEditable && 
          <input type="text" />
        }
      </div>

    );
  }
}

export default KeywordInfo;
