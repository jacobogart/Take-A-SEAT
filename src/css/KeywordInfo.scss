.KeywordInfo {
  width: 190px;
  margin: -20px 0 0 30px;
  background-image: url('../images/paper-background.jpg');
  font-family: $script-font;
  padding: 8px 4px;
  p {
    display: inline-block;
    font-size: 22px;
    margin: 20px 0 0;
    line-height: 25px;
    color: black;
  }
  input {
    margin-bottom: 6px;
    border-color: #758DA3;
    background-color: transparent;
    font-family: "Lucida Console", Monaco, monospace;
    width: 175px;
  }
  button {
    height: 23px;
    margin-top: -2px;
    padding-top: 2px;
    border: 3px solid #966362;;
    border-radius: 10%;
    background-color: #758DA3;
    color: white;
  }
}