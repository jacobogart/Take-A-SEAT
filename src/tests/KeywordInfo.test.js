import React from 'react';
import KeywordInfo from '../components/KeywordInfo';
import { shallow } from 'enzyme';

const mockSelectWord = jest.fn();
const mockKeywordData = {
  "word": "mockData Name",
  "nextWords": [
    "=",
    "}"
  ],
  "value": "mock-",
  "isEditable": true,
  "phase": ["S"],
  "details": "Variable name for your mock data",
  "id": 101
}

describe('KeywordInfo', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <KeywordInfo
        keywordData={mockKeywordData}
        selectWord={mockSelectWord}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke submit value when #keywordForm is submited', () => {
    const mockEvent = {
      preventDefault: () => {},
      target: { querySelector: () => {return { value: ''}} }
    };
    wrapper.find('#keywordForm').simulate("submit", mockEvent);
    expect(mockSelectWord).toHaveBeenCalled();
  });

  it('should invoke selectWord when submitValue is called', () => {
    const mockEvent = {
      preventDefault: () => { },
      target: { querySelector: () => { return { value: '' } } }
    };
    wrapper.instance().submitValue(mockEvent);
    expect(mockSelectWord).toHaveBeenCalled();
  });
    
});