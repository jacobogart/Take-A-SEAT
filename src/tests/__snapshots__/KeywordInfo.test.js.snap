// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordInfo should match snapshot 1`] = `
<div>
  <p>
    Variable name for your mock data
  </p>
  <form
    id="keywordForm"
    onSubmit={[Function]}
  >
    <input
      defaultValue="mock-"
      type="text"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
